### PR TITLE
Chore/upgrade to express

### PR DESCRIPTION
**Pull request by andela-kakpobome**
### What this PR does?
- upgrade the project routing system to fully leverage express
- make endpoints RESTful, start with creating a Fingerprint resource
- migrate current routes to Fingerprints resource
- correct error codes for invalid client requests from `500` to `400`

### CODE REVIEW

File: api/node-echoprint-server/controllers/api.js [hunk 1]
```diff
@@ -1,42 +1,52 @@
...
  */
 exports.query = function(req, res) {
-  var url = urlParser.parse(req.url, true);
-  var code = url.query.code;
-  if (!code)
-    return server.respond(req, res, 500, { error: 'Missing code' });
+  const url = urlParser.parse(req.url, true);
+  const code = url.query.code;
+  if (!code) {
+    res.status(400)
```
Comment (makinwab) [inline api/node-echoprint-server/controllers/api.js:14]: Nice one with the status code.

Comment (andela-kakpobome) [reply api/node-echoprint-server/controllers/api.js:14]: 💪 @andela-bmakinwa

File: api/node-echoprint-server/controllers/api.js [hunk 2]
```diff
@@ -1,42 +1,52 @@
...
+       .send({ error: 'Missing code' });
+  }
 
-  var codeVer = url.query.version;
-  if (codeVer != config.codever)
-    return server.respond(req, res, 500, { error: 'Missing or invalid version' });
+  const codeVer = url.query.version;
```
Comment (makinwab) [inline api/node-echoprint-server/controllers/api.js]: Can we not use abbreviated variable names, we could have something like `codeVersion`.

Comment (makinwab) [reply api/node-echoprint-server/controllers/api.js]: We can do this moving forward though.

Comment (andela-kakpobome) [reply api/node-echoprint-server/controllers/api.js]: @andela-bmakinwa good call, I will update accordingly.

**Comment by makinwab** [comment]
We are closing this PR as [PR6](https://github.com/andela-kakpobome/hazaam/pull/6) has the same changes plus more.

**Pull request closed without merging**